app/src/main/java/com/banuba/quickstart_c_api/GLNVRenderer.java

```diff
@@ -0,0 +1,295 @@
+package com.banuba.quickstart_c_api;
+
+import android.opengl.GLES20;
+import android.opengl.GLES30;
+import android.opengl.GLSurfaceView;
+
+import androidx.annotation.NonNull;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.FloatBuffer;
+
+import javax.microedition.khronos.egl.EGLConfig;
+import javax.microedition.khronos.opengles.GL10;
+
+public class GLNVRenderer implements GLSurfaceView.Renderer {
+    private static final String VERTEX_SHADER_PROGRAM =
+            "#version 300 es\n" +
+                    "precision mediump float;\n" +
+                    "layout (location = 0) in vec3 aPosition;\n" +
+                    "layout (location = 1) in vec2 aTextureCoord;\n" +
+                    "uniform mat4 uMatrix;\n" +
+                    "out vec2 vTexCoord;\n" +
+                    "void main() {\n" +
+                    "  gl_Position = vec4(aPosition, 1.0f) * uMatrix;\n" +
+                    "  vTexCoord = aTextureCoord;\n" +
+                    "}\n";
+
+    private static final String FRAGMENT_SHADER_PROGRAM =
+            "#version 300 es\n" +
+                    "precision mediump float;\n" +
+                    "uniform sampler2D yTexture;\n" +
+                    "uniform sampler2D uvTexture;\n" +
+                    "in vec2 vTexCoord;\n" +
+                    "out vec4 FragColor; \n" +
+                    "void main() \n" +
+                    "{ \n" +
+                        "float r, g, b, y, u, v; \n" +
+                        "y = texture(yTexture, vTexCoord).r; \n" +
+                        "u = texture(uvTexture, vTexCoord).r - 0.5; \n" +
+                        "v = texture(uvTexture, vTexCoord).a - 0.5; \n" +
+                        "float Umax = 0.436; \n" +
```

Comment:
Make sure that you use the right coefficients for conversion - better pass them as a uniform matrix.